README.md

```diff
@@ -52,4 +63,82 @@ Under the hood, the `PairingChannel` implements the "externally-provisioned
 pre-shared key" mode of [TLS1.3](https://tools.ietf.org/html/rfc8446).
 Each side of the channel can thus be assured that its peer is in possession
 of the `channelKey`, and that their traffic is protected from anyone who
-does not possess this key.
+does not possess this key, even from the intermediary channelserver through
+which they exchange messages.
+
+As a consumer of this library, you shouldn't need to care about any of those
+details though - just use the provided APIs to handle all the Crypto bits
+for you.
+
+
+Development and Maintenance
+===========================
+
+This library is basically "done", inasmuch as it works and we are not planning
+any feature additions. We are still maintaining it to keep up-to-date with tooling
+and dependency changes, security fixes, etc.
+
+### Source Code
+
+The main source code for the library is found under `./src`. Since the code gets
+vendored into the Firefox client distribution, we try to avoid taking any external
+dependencies (currently the only dependency is the `event-target-shim` package).
+
+### Tests
+
+There is an extensive test suite under `./test`, covering many of the crypto edge-cases
+that motived the choice to use TLS rather than an ad-hoc protocol. To run them use the
+usual npm test runner:
+
+```
+npm test
+```
+
+This uses the [Karma](https://karma-runner.github.io) test runner to execute
+the tests in a live browser.
+
+Running the tests will also produces coverage reports under `./coverage`,
+which can be viewed with a web browser. We strive for very close to 100%
+test coverage, because this is crypto-related code where even the edgiest
+of edge-cases can have serious consequences if mis-handled.
+
+### Build Artifacts
+
+The main sources get transpiled into two built distribution artifacts, one for use by
+web content and one for vendoring into the Firefox source tree:
+
+* `./dist/FxAccountsPairingChannel.babel.umd.js` is transpiled to ES5-compatible JavaScript
+  using [Babel](https://babeljs.io/), and is suitable for use in web content that must
+  run in any (within reason!) browser. This is the artifact that gets used on the FxA
+  website for the client side of the pairing flow.
+* `./dist/FxAccountsPairingChannel.js` is Firefox browser code, suitable only for use
+  within the Firefox browser application itself. This gets manually copied over into
+  [`mozilla-central/services/fxaccounts/FxAccountsPairingChannel.js`](
+  https://searchfox.org/mozilla-central/source/services/fxaccounts/FxAccountsPairingChannel.js)
+  when making a new release.
+
+To generate these files from the current `./src` tree, run:
+
+```
+npm run build
+```
+
+For ease of consumption, we check the resulting built artifacts in to the Git repo.
+
+### Release Process
+
+Use `npm publish` to cut a new release, after ensuring that the tests pass and the
+correct artifacts have been built:
+
+```
+npm run build
+npm test
+npm publish
+git tag vX.Y.Z & git push origin vX.Y.Z
+```
```

Comment:
I haven't actually tested this release process. We should do so once this batch of PRs land, and tweak as needed.